rfcs/text/0008_pulse.md

```diff
@@ -0,0 +1,247 @@
+- It is indeed a breaking change, but it can be migrated over-time as new features, making use of the instructions.
+- We need to update other products already reporting telemetry from outside Kibana (like Beats, Enterprise Search, Logstash, ...) to use the new way of pushing telemetry.
+
+# Alternatives
+
+> What other designs have been considered?
+
+We currently have the newsfeed to be able to communicate to the user. This is actually pulling in Kibana from a public API to retrieve the list of entries to be shown in the notification bar. But this is only limitted to notifications to the user while the new _intructions_ can provide capabilities like self-update/self-configuration of components like endpoints, elasticsearch, ...
+
+> What is the impact of not doing this?
+
+Users might not see any benefit from providing telemetry and will opt-out. The quality of the telemetry will likely not be as good (or it will require a higher effort on the plugin end to provide it like in [the latest lens effort](https://github.com/elastic/kibana/issues/46599#issuecomment-545024137))
+
+# Adoption strategy
+
+Initially, we'll focus on the remote service and move the current telemetry to report as a `"legacy"` channel to the new Pulse service.
+
+Then, we'll focus on doing the client side, providing new APIs to report the data, aiming for the minimum changes on the public end. For instance, the current usage collectors already report an ID, we can work on those IDs mapping to a channel (only grouping them when it makes sense). Nevertheless, it will require the devs to engage with the Pulse team for the mappings and definitions to be properly set up and updated. And any views to be added.
+
+Finally, the instruction handling APIs are completely new and it will require development on both _remote_ and _local_ ends for the instruction generation and handling.
+
+# How we teach this
+
+> What names and terminology work best for these concepts and why? How is this
+idea best presented? As a continuation of existing Kibana patterns?
+
+We have 3 points of view to show here:
+
+- From the users perspective, we need to show the value for them to have the telemetry activated.
+- From the devs, how to generate data and consume instructions.
+- From the PMs, how to consume the views + definitions of the fields.
+
+> Would the acceptance of this proposal mean the Kibana documentation must be
+re-organized or altered? Does it change how Kibana is taught to new developers
+at any level?
+
+This telemetry is supposed to be internal only. Only internal developers will be able to add to this. So the documentation will only be for internal puposes. As mentioned in the _Adoption strategy_, the idea is that the devs to report new data to telemetry will need to engage with the Pulse team.
```

Comment:
> Only internal developers will be able to add to this.

How will we enforce this? Will we only expose this API to 1st party plugins? Of course, plugins that attempt to inject telemetry with new fields will fail due to the `strict` mapping, but how will we be sure that they can't inject bad/malicious data into the existing fields/channels?

Reply:
That's a fair point! We can open a discussion for it. The current telemetry plugin does not make any distinctions at the moment: Any developer can import the 'usageCollection' plugin and report usage.

So I think we are OK so far with this and can start with an implementation that allows it (still having `strict` mapping will protect us from some unwanted data although doesn't stop devs from injecting it).

Maybe out of the scope for the first phase of the project. But, definitely, a must before we generate any instructions.

Any ideas from the NP team about a way to control what is a _Kibana-original_ plugin vs. external plugins? Maybe it makes sense to expose different core-contexts based on that role?